queries.sql

```diff
@@ -1,15 +1,47 @@
+-- Existing queries
 SELECT * FROM animals WHERE name LIKE '%mon%';
-
 SELECT (name) FROM animals WHERE date_of_birth >= '2016-01-01' AND date_of_birth <= '2019-12-31';
-
 SELECT (name) FROM animals WHERE neutered = true AND escape_attempts < 3;
-
 SELECT (date_of_birth) FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
-
 SELECT (name, escape_attempts) FROM animals WHERE weight_kg > 10.5;
-
 SELECT * FROM animals WHERE neutered = true;
-
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- New transactional queries
+
+-- Step 1: Update species to 'unspecified' and then rollback
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Step 2: Update species based on animal names and commit
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
+-- Step 3: Delete all records and then rollback
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Step 4: Multiple operations with Savepoint
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- New queries for final task
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(*) FROM animals WHERE escape_attempts > 0 GROUP BY neutered;
```

Comment:
👍 Great work on your query! However, there's a slight adjustment needed to match the project requirements:

- [ ] This query currently returns the count of false (2) and true (6) based on how many neutered animals have escape_attempts greater than 0. However, as per the project requirement, you should create a query to find out "Who escapes the most, neutered or not neutered animals?" To achieve this, consider modifying your query to determine whether neutered or not neutered animals have the most escape_attempts. You might want to use functions like `SUM` or `AVG` instead of `COUNT` to calculate this more accurately.

By updating your query in this way, you'll align it with the specific project requirement, ensuring that it provides the intended information. Keep up the great work! 👍📊🐾